tests/cases/fourslash/completionOfAwaitPromiseNoRedundant.ts

```diff
@@ -0,0 +1,21 @@
+/// <reference path='fourslash.ts'/>
+
+//// interface Foo { foo: string }
+//// async function foo(x: (a: number) => Promise<Foo>) {
+////    [|await x(1)./*1*/|]
+////    ;([|await x(1)./*2*/|])
+////    ;(await ([|x(1)./*3*/|]))
```

Comment:
As Jake pointed out, this one seems less than ideal, and very easy to fix. Instead of looking at `propertyAccessToConvert.parent`, just use `skipParentheses(propertyAccessToConvert.parent)`.

Reply:
Sorry, I just saw that you had this and reverted.

> I added support for this (and test), but it is not shown in vscode (because of range of completion)

Can you elaborate on this? Replacement spans have to be on a single line, but I don’t see from your example why that would be a problem.

Reply:
> Also FYI this case is not handled for now (since parent is not await expression):
> 
> ```ts
> (await x(1).foo.test)
> ```

This is fine. This expression already has “excess” parentheses, so it seems correct to add another pair.

Reply:
@andrewbranch Hm, first of all are you sure we want rewrite `await (x(1).foo)` -> `(await x(1)).foo` instead of `await ((await x(1)).foo)` (preserve parens)? This is also the reason of why it doesn't work in vscode.

In vscode [`filterText=insertText`](https://github.com/microsoft/vscode/blob/8e20e0c301a6c81ba8307c78290d3639fbd2efe1/extensions/typescript-language-features/src/languageFeatures/completions.ts#L403), filterText is matched against range. So it sees `(await x(1)).foo` doesn't match `await (x(1).foo)` and doesn't show completion, only if we could change filterText...

P.S. I'm using vscode as easiest way to test changes on real code, have no idea how it works in ts playground (and how to test PRs in ts playground as well).

Reply:
Ah, the `filterText` issue. Whatever. This is already an improvement.